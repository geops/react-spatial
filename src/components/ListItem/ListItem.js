import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * Css class of the li tag.
   */
  className: PropTypes.string,
  /**
   * Item of the list.
   */
  item: PropTypes.any.isRequired,

  /**
   * Children content of the button.
   */
  children: PropTypes.node.isRequired,

  /**
   * Function triggered when you select the item.
   */
  onSelect: PropTypes.func,

  /**
   * Function triggered when you press a key.
   */
  onKeyDown: PropTypes.func,
};

const defaultProps = {
  className: 'tm-list-item',
  onSelect: () => {},
  onKeyDown: () => {},
};

/**
 * This component displays a `<li>` tag.
 */
class ListItem extends PureComponent {
  render() {
    const { className, children, item, onSelect, onKeyDown } = this.props;
    return (
      <li
        className={className}
        role="menuitem"
        tabIndex="0"
        onClick={e => {
          onSelect(e, item);
        }}
        onKeyPress={e => e.which === 13 && onSelect(e, item)}
        onKeyDown={e => {
          onKeyDown(e);
        }}
      >
        {children}
      </li>
    );
  }
}

ListItem.propTypes = propTypes;
ListItem.defaultProps = defaultProps;

export default ListItem;
