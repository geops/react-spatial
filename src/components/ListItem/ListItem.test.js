import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ListItem from './ListItem';

configure({ adapter: new Adapter() });

describe('ListItem', () => {
  describe('when no properties are set', () => {
    let spy = null;

    beforeEach(() => {
      window.console.error = jest.fn().mockImplementation(() => {});
      spy = jest.spyOn(window.console, 'error');
    });

    afterEach(() => {
      spy.mockRestore();
      window.console.error.mockRestore();
    });

    test('displays 1 error for required property ', () => {
      shallow(<ListItem />);
      expect(spy).toHaveBeenCalledTimes(2);
    });

    test('matches snapshot', () => {
      const component = renderer.create(<ListItem />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when properties are set', () => {
    const item = {
      label: 'foo',
    };

    const children = <p>bar</p>;

    test('matches snapshot', () => {
      const component = renderer.create(
        <ListItem item={item} onSelect={() => {}}>
          {children}
        </ListItem>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
