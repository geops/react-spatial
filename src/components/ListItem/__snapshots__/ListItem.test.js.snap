// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem when no properties are set matches snapshot 1`] = `
<li
  className="tm-list-item"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
/>
`;

exports[`ListItem when properties are set matches snapshot 1`] = `
<li
  className="tm-list-item"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <p>
    bar
  </p>
</li>
`;
